probes/requiresUpToDateBranches/def.yml

```diff
@@ -0,0 +1,26 @@
+# Copyright 2023 OpenSSF Scorecard Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+id: requiresUpToDateBranches
+short: Check that the project requires PRs to be in sync with the base branch.
+motivation: >
+  Requiring PRs to be in sync with the base branch is good practice.
```

Comment:
this might need fleshed out a little more.

Reply:
Improved in https://github.com/ossf/scorecard/pull/3691/commits/4071df489be2634f91b81db399f9f38a9cc70841